app/controllers/students_controller.rb

```diff
@@ -17,6 +17,12 @@ def create
   def update
     @student.update(params.require(:student).permit(:approved))
 
+    if @student.approved?
+      @student.term.assignments.each do |assignment|
+        assignment.update(done: false)
```

Comment:
Ok, I understand your idea - there's no need to explicitly update every assignment's status after student's approving.